azext_iot/operations/hub.py

```diff
@@ -1618,6 +1618,41 @@ def iot_hub_configuration_delete(
         handle_service_exception(e)
 
 
+def iot_hub_configuration_test_queries(
+    cmd,
+    target_condition=None,
+    custom_metric_queries=None,
+    hub_name=None,
+    resource_group_name=None,
+    login=None,
+    auth_type_dataplane=None,
+):
+    discovery = IotHubDiscovery(cmd)
+    target = discovery.get_target(
+        resource_name=hub_name,
+        resource_group_name=resource_group_name,
+        login=login,
+        auth_type=auth_type_dataplane,
+    )
+    resolver = SdkResolver(target=target)
+    service_sdk = resolver.get_sdk(SdkType.service_sdk)
+
+    try:
+        if custom_metric_queries:
+            custom_metric_queries = process_json_arg(custom_metric_queries, argument_name="custom_metric_queries")
+
+        result = service_sdk.configuration.test_queries(target_condition, custom_metric_queries)
+
+        if not result.target_condition_error and not result.custom_metric_query_errors:
+            return 'Validation passed!'
+        if result.target_condition_error:
+            return 'Target condition validation failed: {}.'.format(result.target_condition_error)
+        if result.custom_metric_query_errors:
```

Comment:
will there ever be a condition where we have result.target_condition_error and result.custom_metric_query_errors? We want to make sure we return everything if possible

Reply:
there will, let me make a change